hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -299,11 +302,17 @@ private void refreshTimeline() throws IOException {
       schemaProvider = dataAndCheckpoint.getSchemaProvider();
     }
 
-    if ((!avroRDDOptional.isPresent()) || (avroRDDOptional.get().isEmpty())) {
-      log.info("No new data, nothing to commit.. ");
+    if (Objects.equals(checkpointStr, resumeCheckpointStr.orElse(null))) {
+      log.info("No progress in source consumption. Old checkeckpoint=(" + resumeCheckpointStr
+          + "). New Checkpoint=(" + checkpointStr + ")");
       return null;
     }
 
+    if ((!avroRDDOptional.isPresent()) || (avroRDDOptional.get().isEmpty())) {
+      log.info("No new data, return empty rdd.. ");
```

Comment:
Change to : "No new data, perform empty commit"

Reply:
Done